sqlmesh/core/lineage.py

```diff
@@ -15,44 +16,63 @@
     from sqlmesh.core.model import Model
 
 
-def _render_query(model: Model) -> exp.Query:
-    """Render a model's query, adding in managed columns"""
-    query = model.render_query_or_raise().copy()
-    if model.managed_columns:
-        query.select(
-            *[
-                exp.alias_(exp.cast(exp.Null(), to=col_type), col)
-                for col, col_type in model.managed_columns.items()
-                if col not in query.named_selects
-            ],
-            append=True,
-            copy=False,
-        )
-    return query
+CACHE: t.Dict[str, t.Tuple[int, exp.Expression, Scope]] = {}
```

Comment:
Should this be global? Can it ever be modified concurrently & if so, would there be issues?

Reply:
in general our code is not really thread safe, so this is fine i think